wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -564,14 +589,27 @@ public Revision commit(CommitParams params) throws GitException {
             }
 
             CommitCommand commitCommand = getGit().commit()
-                                                  .setCommitter(committerName, committerEmail).setAuthor(committerName, committerEmail)
+                                                  .setAllowEmpty(params.isAmend())
+                                                  .setCommitter(committerName, committerEmail)
+                                                  .setAuthor(committerName, committerEmail)
                                                   .setMessage(message)
                                                   .setAll(params.isAll())
                                                   .setAmend(params.isAmend());
 
-            if (!params.isAll()) {
-                params.getFiles().forEach(commitCommand::setOnly);
-            }
+            /*
+            When committing  with amend flag specified paths that don't have any changes, JGitInternalException will be thrown.
+            According to setAllowEmpty method documentation, setting this flag to true must allow such commit,
+            but it won't because Jgit has a bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=510685.
+            As a workaround, add only changed and specified paths to specified paths of the commit command, and throw exception,
+            if other changes are present, but the list of changed and specified paths is empty, to prevent committing other paths.
+            TODO Remove throwing exception when the bug will be fixed.
+            */
+            if (!specified.isEmpty() && !(params.isAll() ? changed.isEmpty() : staged.isEmpty()) && specifiedChanged.isEmpty()) {
```

Comment:
Please double check that it is right check. Looks like we should throw exception [here ](https://github.com/eclipse/che/pull/3833/files#diff-7f323c0167c366f41270f82a35ed368dR569) in case when specified is not empty and specifiedChanged is empty.